src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitionInfo.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.PowerFx.Core.Parser;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Syntax.SourceInformation;
+
+internal class UserDefinitionInfo
+{
+    internal int Index;
+    internal UserDefinitionType Type;
+    internal IdentToken Name;
+    internal string Declaration;
+    internal string Script;
+    internal SourceList Before;
+    internal SourceList After;
+
+    internal UserDefinitionInfo(int index, UserDefinitionType type, IdentToken name, string declaration, SourceList before, SourceList after = null)
+    {
+        Index = index;
+        Type = type;
+        Name = name;
+        Declaration = declaration;

```

Comment:
This is a stopgap while we move over to the unified parser. The only part that gets properly parsed is the definition body so this is a "hack" to save all comments that are made in the declaration of the UDF/NF/type.

Reply:
For the most part this formatter change will format the function body only, and put each definition on its own newline.